### ISSUE TITLE
Decrease the latency of pulldown from sycl branch to sycl-mlir branch

### ISSUE THREAD

**Issue by keryell** [intel/llvm#12047]
There is currently a pulldown of upstream LLVM into `sycl` branch around twice a week, which is great.
Unfortunately, the pulldown from the `sycl` branch into `sycl-mlir` happens quite less often.
This makes difficult the mix-and-match of this project with other MLIR projects.

**Comment by victor-eds** [comment]
Note we have a bot that should be handling this automatically, but it's not working lately... I'll try to get this back on track on Monday to make the whole process smoother.

**Comment by sommerlukas** [comment]
Hi @keryell,

thanks for your interest in the project!

There is a bot to automate the pulldown from `sycl` to `sycl-mlir`, which works fine as long as there are no merge conflicts. 

If there is a merge conflict, these need to be resolved manually, which we currently do on a best-effort basis. 

If you have a specific MLIR/LLVM feature that you need in a short timeframe, you can follow the instructions in #12050 for the pulldown. We are happy to review PRs for pulldowns. 

With the typed pointer support now (virtually) removed from upstream MLIR, we hope that we will encounter merge conflicts less frequently as soon as we have removed the last remainders of that API (we have already switched to opaque pointers a while ago) and the automatic bot would lead to a decreased latency.

**Comment by keryell** [comment]
Thank you for updating the documentation. If you are motivated, updating the top-level README of this MLIR project could be useful for the future generations. :-)
I am struggling just to have the baseline of the tests passing.
https://github.com/intel/llvm/pull/12087
https://github.com/intel/llvm/pull/12090
https://github.com/intel/llvm/issues/12091

**Comment by victor-eds** [comment]
> If you are motivated, updating the top-level README of this MLIR project could be useful for the future generations. :-)

Yes, maybe having some kind of pointer to the SYCL-MLIR README would be a good step forward. Thanks, @keryell!
